docs/HACKING.md

```diff
@@ -112,47 +142,293 @@ Additional flags that can be passed to `hack/run-ci-e2e-test.sh` are
+| 2019-Datacenter-with-Containers | mcr.microsoft.com/powershell:lts-nanoserver-1809     |
+| 2022-datacenter-smalldisk       | mcr.microsoft.com/powershell:lts-nanoserver-ltsc2022 |
+| datacenter-core-20h2-with-containers-smalldisk | mcr.microsoft.com/powershell:lts-nanoserver-20h2|
+
+If you are using the example deployment, a sed command can update the base image for you.
+
+```shell script
+export BASE_IMAGE=powershell:lts-nanoserver-ltsc2022
+sed -i 's/image:.*$/image: "mcr.microsoft.com\/'$BASE_IMAGE'"/' deployment.yaml
+```
+
+##### Set the powershell executable
+
+Different base images have different executables for powershell. Since WMCO uses the powershell nanoserver images, the
+command should be set to pwsh.exe.
+Change the executable to pwsh.exe by running
+
+```shell script
+sed -i "s/powershell.exe/pwsh.exe/" deployment.yaml
+```
+
+### Deploy the winworker machine
+
+In order to generate and apply the windows MachineSet.yaml you must enter the top level of your wmco directory and run
+
+```shell script
+hack/machineset.sh apply
+```
+
+### Deploy the workload
+
+now you can deploy the workload by running
+
+```shell script
+oc apply -f deployment.yaml
+```
+
+### Cleanup custom workload
+
+```shell script
+oc delete deployment win-webserver
+hack/olm.sh cleanup
+oc delete ns wmco-test
+```
+
+## Manual build process
+
+## Build an operator image
+
+If you want to deploy the operator manually without the OLM, you need to build an operator image that we can deploy to
+the cluster.
+From the top level of your wmco directory, run
+
+```shell script
+podman build -t quay.io/<quay username>/wmco:<operator name> -f build/Dockerfile .
+```
+
+Building the full WMCO image is a time consuming process given we are building all the submodules. If you are just
+changing WMCO and not the submodules, you can separately build the submodules as a base image and then just build the
+WMCO  image from this base image:
+
+```shell script
+make base-img
+make wmco-img IMG=$OPERATOR_IMAGE
+```
+
+`make base-img` will not push the base image as a push is not needed. `make wmco-img` will make and push the
+`$OPERATOR_IMAGE`
+
+### Push an operator image
+
+Push your image to your container registry with podman
+
+`podman push quay.io/<quay username>/wmco:<operator name>`
+
+### Deploying the operator directly
+
+```shell script
+make deploy IMG=$OPERATOR_IMAGE
+```
+
+The above command will directly deploy all the resources required for the operator to run, as well as the WMCO pod.
+To remove the installed resources:
+
+```shell script
+make undeploy
+```
+
+### Run Tests
+
+Now that the WMCO is deployed, see the above instructions for running [e2e tests](#running-e2e-tests) or
+[deploying custom workloads](#deploying-a-custom-workload)
+
+## Updating submodule versions
+
+This project contains git submodules for the following components:
+
+- kubernetes
+- ovn-kubernetes
+- containernetworking-plugins
+- promu
+- windows_exporter
```

Comment:
please add containerd, hcsshim, kube-proxy, and cloud-provider-azure to this list